### PR TITLE
Set Default Song in Levelbuilder

### PR DESCRIPTION
**Pull request by epeach**
Builds on: https://github.com/code-dot-org/code-dot-org/pull/25058 and https://github.com/code-dot-org/code-dot-org/pull/25081

*Adds dropdown in levelbuilder to set the default song
*Song Selection UI is auto-populated with this value
*Note: The dropdown is populated from a manually curated list of songs in dancelab.rb I left a todo in there for myself as a future step will be the use the song library manifest to populate these values.

Has no change on user-experience, so not behind the 'songSelector' flag.

Upcoming PRs:
Use the song chosen in the Song Selector UI when the project is played
Remove set song block

### CODE REVIEW

File: dashboard/app/views/levels/editors/_blockly.html.haml [hunk 1]
```diff
@@ -70,6 +70,11 @@
   #plusPreloadAssetList
     %i.fa.fa-plus-circle
 
+.field
+  = f.label :default_song, 'Dance Lab Only - Default Song'
+  = f.select :default_song, options_for_select(@level.class.hoc_songs, @level.default_song)
```
Comment (davidsbailey) [inline dashboard/app/views/levels/editors/_blockly.html.haml]: better to wrap this `.field` with `- if @level.is_a? Dancelab`.

the next time we add a setting specific to Dancelab, we should really extract a new _dancelab.html.haml partial, and then refer to it like this:  https://github.com/code-dot-org/code-dot-org/blob/66deb81a3be1ff69bb46948ee2c950aa436b50b0/dashboard/app/views/levels/_form.html.haml#L43

File: dashboard/app/models/levels/level.rb [hunk 2]
```diff
@@ -187,6 +187,10 @@ def self.flower_types
   def self.palette_categories
   end
 
+  # Overriden by different level types.
+  def self.hoc_songs
+  end
+
```
Comment (davidsbailey) [inline dashboard/app/models/levels/level.rb]: ok to omit this method definition once you are ensuring that it is only called for Dancelab.

File: dashboard/app/models/levels/blockly.rb [hunk 3]
```diff
@@ -79,6 +79,7 @@ class Blockly < Level
     preload_asset_list
     skip_autosave
     skip_run_save
+    default_song
```
Comment (joshlory) [inline dashboard/app/models/levels/blockly.rb]: Will other app types use this in the future, or should this go in gamelab.rb?

File: dashboard/app/helpers/levels_helper.rb [hunk 4]
```diff
@@ -534,6 +534,10 @@ def blockly_options
       level_options[:sharedFunctions] = nil # TODO: handle non-standard pools
     end
 
+    if @level.is_a? Dancelab
+      level_options[:defaultSong] = @level.default_song
+    end
```
Comment (joshlory) [inline dashboard/app/helpers/levels_helper.rb]: I don't think we need this — the following ensures that all level properties are turned into options on `appOptions.level`:

https://github.com/code-dot-org/code-dot-org/blob/cda7db3f6fdcdc68781220d53735f40b3215b52d/dashboard/app/helpers/levels_helper.rb#L503-L505

**Pull request merged**